src/content_sao.cpp

```diff
@@ -1273,6 +1273,42 @@ std::string PlayerSAO::getWieldList() const
 	return "main";
 }
 
+ItemStack PlayerSAO::getWieldedItem() const
+{
+	const Inventory *inv = getInventory();
+	ItemStack ret;
+	if (inv) {
+		const InventoryList *mlist = inv->getList(getWieldList());
+		if (mlist && (getWieldIndex() < (s32)mlist->getSize()))
+			ret = mlist->getItem(getWieldIndex());
+		if (ret.name == "") {
+			const InventoryList *hlist = inv->getList("hand");
+			if (hlist)
+				ret = hlist->getItem(0);
+		}
+	}
+	return ret;
+}
+bool PlayerSAO::setWieldedItem(const ItemStack &item)
```

Comment:
No. setWieldedItem is used for the wielded item, not only for the hand. I need to override it, because else, you might get a replacement for the hand in your inventory, if you use a hand that changes when used.